bench/tx-generator/src/Cardano/Benchmarking/Script/Core.hs

```diff
@@ -632,6 +643,37 @@ createChangeGeneric submitMode createCoins addressMsg value count = do
   chunkList _ [] = []
   chunkList n xs = as : chunkList n bs where (as,bs) = splitAt n xs
 
+spendAutoScript :: SubmitMode -> FilePath -> ThreadName -> NumberOfTxs -> TPSRate -> ActionM ()
+spendAutoScript submitMode scriptFile threadName txCount tps = do
+  protocolParameters <- queryProtocolParameters
```

Comment:
By the way, the initial version of `cardano-node` started with this pattern of passing information through the underlying monad.

In Byron Rewrite we dropped this in favor of explicit parameter passing -- because it makes the data dependencies easier to understand.

Reply:
This is not really the underlying monad though.
This is a real/full query for the protocol parameters of the locally connected node.

Reply:
Oh, that explains -- so at the start of a particular benchmark the parameters are always up to date.

That's actually a useful feature, thank you for explaining Marc!